insta/src/runtime.rs

```diff
@@ -51,43 +55,86 @@ macro_rules! elog {
 #[derive(Debug)]
 pub struct AutoName;
 
-impl From<AutoName> for ReferenceValue<'static> {
-    fn from(_value: AutoName) -> ReferenceValue<'static> {
-        ReferenceValue::File(None)
+pub struct InlineValue<'a>(pub &'a str);
+
+impl From<AutoName> for SnapshotName<'static> {
+    fn from(_value: AutoName) -> SnapshotName<'static> {
+        None
     }
 }
 
-impl From<Option<String>> for ReferenceValue<'static> {
-    fn from(value: Option<String>) -> ReferenceValue<'static> {
-        ReferenceValue::File(value.map(Cow::Owned))
+type SnapshotName<'a> = Option<Cow<'a, str>>;
+
+pub enum SnapshotValue<'a> {
+    Text {
```

Comment:
To the extent we keep this enum — the naming isn't ideal, where `Text` includes `Inline` elsewhere but not here?